File: BilDunya/src/main/java/com/bildunya/dto/ModerateContentRequest.java
package com.bildunya.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ModerateContentRequest {

    @NotBlank(message = "verificationStatus is required")
    private String verificationStatus; // VERIFIED, REJECTED

    @Size(max = 500, message = "rejectionReason must be at most 500 characters")
    private String rejectionReason;
}

File: bildunya_frontend/lib/data/models/moderate_content_request.dart
/// `PATCH /contents/{id}/moderation` gövdesi.
class ModerateContentRequest {
  const ModerateContentRequest({
    required this.verificationStatus,
    this.rejectionReason,
  });

  final String verificationStatus; // APPROVED, REJECTED
  final String? rejectionReason;

  Map<String, dynamic> toJson() => {
    'verificationStatus': verificationStatus,
    if (rejectionReason != null) 'rejectionReason': rejectionReason,
  };
}
